airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -37,6 +45,10 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationRunner.class);
 
+  public static final int INTERRUPT_THREAD_DELAY_MINUTES = 60;
+  public static final int EXIT_THREAD_DELAY_MINUTES = 4 * 60;
```

Comment:
These time-outs are triggered after the moment the main thread is exiting its main loop before we start nudging or forcing the remaining threads to stop